contrib/opentimelineio_contrib/application_plugins/tests/test_rv_reader.py

```diff
@@ -0,0 +1,246 @@
+    "OTIO_RV_ROOT_DIR not set."
+)
+@unittest.skipIf(
+    (sys.version_info > (3, 0)),
+    "RV OTIO reader plugin does not work in python 3."
+)
+class RVSessionAdapterReadTest(unittest.TestCase):
+    def create_temp_dir(self):
+        return tempfile.mkdtemp(prefix='rv_otio_reader')
+
+    def test_create_rvpkg(self):
+        temp_dir = self.create_temp_dir()
+        package_path = create_rvpkg(temp_dir)
+
+        self.assertTrue(os.path.exists(package_path))
+        self.assertTrue(os.path.getsize(package_path) > 0)
+
+        # Cleanup
+        shutil.rmtree(temp_dir)
+
+    def test_install_plugin(self):
+        temp_dir = self.create_temp_dir()
+        source_package_path = create_rvpkg(temp_dir)
+
+        # Install package
+        rc = install_package(source_package_path)
+
+        # Check if install succeeded
+        installed_package_path = os.path.join(
+            temp_dir,
+            'Packages',
+            os.path.basename(source_package_path)
+        )
+
+        self.assertTrue(rc == 0)
+        self.assertTrue(len(os.listdir(temp_dir)) > 1)
+        self.assertTrue(os.path.exists(installed_package_path))
+
+        # Make sure package is available in RV
+        list_cmd = '{root}/rvpkg ' \
+                   '-only {tmp_dir} -list'\
+                   .format(
+                       root=RV_BIN_DIR,
+                       tmp_dir=temp_dir
+                   )
+
+        proc = Popen(shlex.split(list_cmd), stdout=PIPE)
+        stdout, _ = proc.communicate()
+
+        desired_result = \
+            'I L - {version} "{package_name}" {pkg_path}'.format(
+                version=RV_OTIO_READER_VERSION,
+                package_name=RV_OTIO_READER_NAME,
+                pkg_path=installed_package_path
+            )
+
+        self.assertIn(desired_result, stdout.split('\n'))
+
+        # Cleanup
+        shutil.rmtree(temp_dir)
+
+    def test_read_otio_file(self):
+        # Install package
+        temp_dir = self.create_temp_dir()
+        source_package_path = create_rvpkg(temp_dir)
+        install_package(source_package_path)
+
+        env = os.environ.copy()
+        env.update({'RV_SUPPORT_PATH': temp_dir})
+
+        sample_file = tempfile.NamedTemporaryFile(
+            'w',
+            prefix='otio_data_',
+            suffix='.otio',
+            dir=temp_dir,
+            delete=False
+        )
+        otio.adapters.write_to_file(sample_timeline, sample_file.name)
+        run_cmd = '{root}/rv ' \
+                  '-nc ' \
+                  '-network ' \
+                  '-networkHost localhost ' \
+                  '-networkPort {port} ' \
+                  '{sample_file}' \
+                  .format(
+                      root=RV_BIN_DIR,
+                      port=9876,
+                      sample_file=sample_file.name
+                  )
+        proc = Popen(shlex.split(run_cmd), env=env)
+
+        # Dirty way to wait for RV to launch
+        time.sleep(4)
```

Comment:
How could we make this more robust?  I'd like to believe that all RVs would run in less than 4 seconds, but would some sort of semaphore be more reliable?

Reply:
Good point. I'll look at some options. Thanks!